### PR TITLE
Test to confirm the status of two containers within a pod  post migration

### PR DESCRIPTION
**Pull request by DeepthiYV**
Run two containers within a pod under systemd,  ensure the status of containers running after the migration of SLEM 5.5 to SLEM6.0

- Related ticket: https://progress.opensuse.org/issues/155857
- Needles:NO
- Verification run: http://deepthi-openqa.qe.suse.de/tests/6923#

### CODE REVIEW

File: tests/containers/testenv_prepare.pm [hunk 1]
```diff
@@ -23,31 +23,44 @@ use containers::container_images;
...
+    systemctl("is-active pod-test-pod0.service");
+
+    # Start 2 containers service and ensure its running
+    assert_script_run("systemctl enable --now container-Tumbleweed-container.service", timeout => 120);
+    systemctl("is-active container-Tumbleweed-container.service");
+
+    assert_script_run("systemctl enable --now container-nginx-container.service", timeout => 120);
+    systemctl("is-active container-nginx-container.service");
+    # Verify the connection between containers in a pod
+    validate_script_output("podman exec -it Tumbleweed-container /bin/bash -c 'curl -s http://localhost:80 | grep title'", sub { m/Welcome to the nginx container!/ });
```
Comment (ricardobranco777) [inline tests/containers/testenv_prepare.pm:63]: Perhaps easier:

```suggestion
    validate_script_output("podman exec -it Tumbleweed-container curl -s http://localhost:80", sub { m/Welcome to the nginx container!/ });
```

File: tests/containers/testenv_prepare.pm [hunk 2]
```diff
@@ -23,31 +23,44 @@ use containers::container_images;
...
     $self->{podman} = $podman;
 
-    # create files for generator
-    record_info("quadlet version", script_output("$quadlet -version"));
-    record_info('Unit', script_output("$quadlet -v -dryrun"));
+    # create a pod with 2 containers in it
+    assert_script_run("podman pod create --name test-pod0 -p 80:80");
 
-    # start the generator and check whether the files are generated
+    assert_script_run("podman run -d --name nginx-container --pod  test-pod0  -v  /home/nginx/nginx.conf:/etc/nginx/nginx.conf:ro,z  -v /home/nginx/index.html:/usr/share/nginx/html/index.html:ro,z  $nginx_container_image");
```
Comment (ricardobranco777) [inline tests/containers/testenv_prepare.pm:40]:
```suggestion
    assert_script_run("podman run -d --name nginx-container --pod test-pod0 -v /home/nginx/nginx.conf:/etc/nginx/nginx.conf:ro,z -v /home/nginx/index.html:/usr/share/nginx/html/index.html:ro,z $nginx_container_image");
```

File: tests/containers/testenv_validation.pm [hunk 3]
```diff
@@ -28,11 +28,17 @@ sub run {
...
+    record_info('podman pod ps', script_output("podman pod ps"));
+    record_info('podman ps', script_output("podman ps"));
+
+    # Ensure the pod and containers are running
+    systemctl("is-active pod-test-pod0.service");
+    systemctl("is-active container-Tumbleweed-container.service");
+    systemctl("is-active container-nginx-container.service");
+
+    # Verify the connection between containers in a pod
+    validate_script_output("podman exec -it Tumbleweed-container /bin/bash -c 'curl -s http://localhost:80 | grep title'", sub { m/Welcome to the nginx container!/ });
```
Comment (ricardobranco777) [inline tests/containers/testenv_validation.pm:41]: Ditto above.

File: tests/containers/testenv_prepare.pm [hunk 4]
```diff
@@ -23,31 +23,44 @@ use containers::container_images;
...
     my ($self) = @_;
-    my $unit_name = 'test_nginx';
 
     select_serial_terminal;
-    my $quadlet = script_output "rpm -ql podman | grep podman/quadlet";
+    my $tumbleweed_container_image = "registry.opensuse.org/opensuse/tumbleweed:latest";
+    my $nginx_container_image = "registry.opensuse.org/opensuse/nginx:latest";
 
     assert_script_run('curl -sLf --create-dirs -vo /home/nginx/nginx.conf ' . data_url('containers/nginx/') . 'nginx.conf');
     assert_script_run('curl -sLf --create-dirs -vo /home/nginx/index.html ' . data_url('containers/nginx/') . 'index.html');
```
Comment (m-dati) [inline tests/containers/testenv_prepare.pm:32]:
```suggestion
    assert_script_run('curl -sSLf --create-dirs -vo /home/nginx/index.html ' . data_url('containers/nginx/') . 'index.html');
```
It is ok now, only as simple suggestion: the `-s` option is to remove all ouptput; by adding `-S` we show eventual errors instead, to ease debugging.

File: tests/containers/testenv_validation.pm [hunk 5]
```diff
@@ -28,11 +28,17 @@ sub run {
 
     # validate the container
     my $target_version = get_var("TARGET_VERSION", get_required_var("VERSION"));
-    validate_script_output("cat /etc/os-release", sub { m/VERSION="$target_version"/ });
-    validate_script_output("podman ps", qr/$container_name/);
-    validate_script_output("podman container inspect --format='{{.State.Running}}' $container_name", qr/true/);
-    systemctl("is-active $unit_name.service");
-    validate_script_output("curl -s http://localhost:80 | grep title", sub { m/Welcome to the nginx container!/ });
+    validate_script_output('podman pod ps', sub { m/test-pod0/ });
```
[the remark this answers is not in the document]
Comment (DeepthiYV) [reply tests/containers/testenv_validation.pm:31]: I created a pod with the name `test-pod0`, so I am checking it here.

**Pull request merged**